backend/dal/Services/Admin/Concrete/UserService.cs

```diff
@@ -237,6 +237,42 @@ public Paged<AccessRequest> GetAccessRequests(int page = 1, int quantity = 10, s
             var accessRequests = query.Skip((page - 1) * quantity).Take(quantity);
             return new Paged<AccessRequest>(accessRequests, page, quantity, query.Count());
         }
+
+        /// <summary>
+        /// Get all the access requests that users have submitted to the system
+        /// </summary>
+        /// <param name="filter"></param>
+        public Paged<AccessRequest> GetAccessRequests(AccessRequestFilter filter)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.AdminUsers);
+
+            var query = this.Context.AccessRequests
+                .Include(p => p.Agencies)
+                .ThenInclude(p => p.Agency)
+                .Include(p => p.Roles)
+                .ThenInclude(p => p.Role)
+                .Include(p => p.User)
+                .AsNoTracking();
+
+            query = query.Where(request => request.Status == filter.Status);
+
+            if (!string.IsNullOrWhiteSpace(filter.Role))
+                query = query.Where(ar => ar.Roles.Any(r =>
+                    EF.Functions.Like(r.Role.Name, $"%{filter.Role}")));
+
+            if (!string.IsNullOrWhiteSpace(filter.Agency))
+                query = query.Where(ar => ar.Agencies.Any(a =>
+                    EF.Functions.Like(a.Agency.Name, $"%{filter.Agency}")));
+
+            if (!string.IsNullOrWhiteSpace(filter.SearchText))
+            {
+                query = query.Where(ar => EF.Functions.Like(ar.User.Username, $"%{filter.SearchText}"));
+            }
```

Comment:
I'm okay with these filter but would like to ask @Fosol whether it makes sense to have these filter be "start-with". I think some of them might not find data that is there - specially the search text one. 

The user would have to know that they are doing a "start-with" query, instead of a "contains"

Thoughts?

Reply:
I made a mistake here, changing it to `$"%{filter.SearchText}%"`